### PR TITLE
Address the remaining comments on PR #6

### PR DESCRIPTION
**Pull request by elgnay**
This PR addressed the remaining comments on PR #6

### CODE REVIEW

File: pkg/spoke/hubclientcert/certificate.go [hunk 1]
```diff
@@ -86,7 +86,7 @@ func getCertValidityPeriod(secret *corev1.Secret) (*time.Time, *time.Time, error
 
 	certs, err := certutil.ParseCertsPEM(certData)
 	if err != nil {
-		return nil, nil, fmt.Errorf("unable to parse TLS certificates: %v", err)
+		return nil, nil, fmt.Errorf("unable to parse TLS certificates: %w", err)
```
Comment (pmorie) [inline pkg/spoke/hubclientcert/certificate.go:89]: Why change this? `%w` will just be a synonym for `%v` here

File: pkg/spoke/spokeagent.go [hunk 2]
```diff
@@ -69,15 +70,19 @@ func (o *SpokeAgentOptions) RunSpokeAgent(ctx context.Context, controllerContext
...
 	// create kube client for spoke cluster
 	spokeKubeClient, err := kubernetes.NewForConfig(controllerContext.KubeConfig)
 	if err != nil {
 		return err
 	}
 
+	// create an informer for secrets on spoke cluster
+	spokeSecretInformer := informers.NewSharedInformerFactory(spokeKubeClient,
+		10*time.Minute).Core().V1().Secrets()
+	go spokeSecretInformer.Informer().Run(ctx.Done())
```
Comment (pmorie) [inline pkg/spoke/spokeagent.go:84]: nit: I think the `Run` invocation could move next to where the other informer and controller are run

**Comment by pmorie** [comment]
Just a couple nits, lgtm otherwise

File: pkg/spoke/hubclientcert/certificate.go (hunk 1 above)
Comment (deads2k) [reply pkg/spoke/hubclientcert/certificate.go:89]: > Why change this? `%w` will just be a synonym for `%v` here

it allows determining chained types, right?

Comment (deads2k) [reply pkg/spoke/hubclientcert/certificate.go:89]: > Why change this? `%w` will just be a synonym for `%v` here

see the wrappping errors bits of https://blog.golang.org/go1.13-errors

**Comment by deads2k** [comment]
/lgtm
/hold

holding for @pmorie's request

File: pkg/spoke/hubclientcert/certificate.go (hunk 1 above)
Comment (pmorie) [reply pkg/spoke/hubclientcert/certificate.go:89]: Interesting, I was not aware of this, disregard my comments re: `%w`

**Pull request merged**